test/shopping-list.test.js

```diff
@@ -0,0 +1,151 @@
+  });
+
+  it("should have a method named 'uncheck'", function () {
+    const pizza = new ShoppingListItem('pizza', 'pepperoni');
+
+    expect(pizza.uncheck).to.exist;
+    expect(pizza.uncheck).to.be.a('function');
+
+    pizza.check();
+    pizza.uncheck();
+    expect(pizza.isDone).to.deep.equal(false);
+  });
+
+  it("should have a method named 'render'", function () {
+    const pizza = new ShoppingListItem('pizza', 'pepperoni');
+
+    expect(pizza.render).to.exist;
+    expect(pizza.render).to.be.a('function');
+    expect(pizza.render()).to.deep.equal(
+      '<li class="completed_false"><span>pizza</span> <span>pepperoni</span></li>'
+    );
+  });
+});
+
+describe('ShoppingList', function () {
+  it('should be a class', function () {
+    expect(ShoppingList).to.be.a('function');
+    expect(new ShoppingList()).to.be.instanceof(ShoppingList);
+  });
+
+  it("should have a property named 'items'", function () {
+    expect(new ShoppingList()).to.have.property('items');
+    expect(new ShoppingList().items).to.be.an('array');
+    expect(new ShoppingList().items).to.deep.equal([]);
+  });
+
+  it("should have a constructor method that initializes 'items' as an empty array", function () {
+    const newList = new ShoppingList('apple', 'banana');
+
+    expect(new ShoppingList().items).to.deep.equal([]);
+    expect(newList.items).to.deep.equal([]);
+  });
+
+  it("should have a method named 'addItem' that accepts a single ShoppingListItem argument", function () {
+    const newList = new ShoppingList();
+    const apple = new ShoppingListItem('apple', 'red fruit');
+
+    expect(newList.addItem).to.exist;
+    expect(newList.addItem).to.be.a('function');
+
+    newList.addItem(apple);
+    expect(newList.items).to.deep.equal([apple]);
+
+    expect(newList.addItem.bind(newList, 'daslkjasd')).to.throw(
+      'addItem only adds shopping list items'
+    );
+  });
+
+  it("should have a method named 'removeItem' that accepts a single ShoppingListItem argument", function () {
+    const newList = new ShoppingList();
+    const apple = new ShoppingListItem('apple', 'red fruit');
+    const pizza = new ShoppingListItem('pizza', 'pepperoni');
+
+    newList.addItem(apple);
+    newList.removeItem(apple);
+    expect(newList.items).to.not.contain(apple);
+
+    newList.addItem(apple);
+    newList.addItem(pizza);
+    newList.removeItem();
+    expect(newList.items).to.not.contain(pizza);
+
+    newList.items = [];
+    newList.removeItem();
+    expect(newList.items).to.deep.equal([]);
+
+    expect(newList.removeItem.bind(newList, pizza)).to.throw(
+      `could not remove item because it was not found in the list`
+    );
+  });
+
+  it("should have a method named 'render'", function () {
+    const newList = new ShoppingList();
+    const apple = new ShoppingListItem('apple', 'red fruit');
+    const pizza = new ShoppingListItem('pizza', 'pepperoni');
+
+    expect(newList.render).to.exist;
+    expect(newList.render).to.be.a('function');
```

Comment:
There's a respondsTo assertion in chai that checks if a method exists on a class.